delfin/drivers/hds/vsp/vspstor.py

```diff
@@ -0,0 +1,266 @@
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rest_client = RestClient(**kwargs)
+        self.rest_handler = rest_handler.RestHandler(self.rest_client)
+        self.rest_handler.get_device_id()
+        self.rest_handler.login()
+
+    def reset_connection(self, context, **kwargs):
+        self.rest_handler.logout()
+        self.rest_client.verify = kwargs.get('verify', False)
+        self.rest_handler.get_device_id()
+        self.rest_handler.login()
+
+    def close_connection(self):
+        self.rest_handler.logout()
+
+    def get_storage(self, context):
+        free_total = 0
+        total_total = 0
+        used_total = 0
+        self.rest_handler.get_device_id()
+        if self.rest_handler.DEVICE_MODEL in consts.VSP_MODEL_NOT_USE_SVPIP:
+            capacity_json = self.rest_handler.get_capacity()
+            free_total = capacity_json.get("total").get("freeSpace") * units.Ki
+            total_total = capacity_json.get("total").get("totalCapacity") * \
+                units.Ki
+            used_total = total_total - free_total
+        else:
+            pools_info = self.rest_handler.get_all_pools()
+            if pools_info is not None:
+                pools = pools_info.get('data')
+                for pool in pools:
+                    total_cap = \
+                        int(pool.get(
+                            'totalPoolCapacity')) * units.Mi
+                    free_cap = int(
+                        pool.get(
+                            'availableVolumeCapacity')) * units.Mi
+                    free_total = free_total + free_cap
+                    total_total = total_total + total_cap
+                used_total = total_total - free_total
+        firmware_version = self.rest_handler.get_specific_storage()
+        status = constants.StorageStatus.OFFLINE
+        if firmware_version is not None:
+            status = constants.StorageStatus.NORMAL
+        system_name = '%s_%s' % (self.rest_handler.DEVICE_MODEL,
+                                 self.rest_client.rest_host)
+
+        s = {
+            'name': system_name,
+            'vendor': 'Hitachi',
+            'description': 'Hitachi VSP Storage',
+            'model': str(self.rest_handler.DEVICE_MODEL),
+            'status': status,
+            'serial_number': str(self.rest_handler.SERIALNUMBER),
+            'firmware_version': str(firmware_version),
+            'location': '',
+            'raw_capacity': int(total_total),
+            'total_capacity': int(total_total),
+            'used_capacity': int(used_total),
+            'free_capacity': int(free_total)
+        }
+        return s
+
+    def list_storage_pools(self, context):
+        try:
+            pools_info = self.rest_handler.get_all_pools()
+            pool_list = []
+            if pools_info is not None:
+                pools = pools_info.get('data')
+            else:
+                return pool_list
```

Comment:
This branch meanings some error during `get_all_pools()`? Or some other means?
I'm concerning about that, if I got a empty list from `list_storage_pools()`, it means actually there is no pool or met some issue during querying pool info?
Same for `list_volumes`

Reply:
yes,there may be met some issue during querying pool info,so i return a empty list

Reply:
So when I get an empty list from `list_storage_pools`, how can I know whether it has no pool or it has pools but I met some issue when querying?

Reply:
changed